mir_restapi/mir_restapi/mir_restapi_lib.py

```diff
@@ -0,0 +1,214 @@
+            if print:
+                self.logger.info("REST API: Connected!")
+        except Exception as e:
+            if print:
+                self.logger.warn('REST API: Attempt to connect failed: ' + str(e))
+            return False
+        return True
+
+    def is_available(self):
+        status = json.dumps(self.get_status())
+        if "service_unavailable" in status:
+            return False
+        else:
+            return True
+    
+    def wait_for_available(self):
+        while True:
+            if self.is_connected(print=False):
+                if self.is_available():
+                    self.logger.info('REST API: available')
+                    break
+                else:
+                    self.logger.info('REST API: unavailable... waiting')
+                    time.sleep(1)
+
+    def get_status(self):
+        response = self.http.get("/status")
+        return json.loads(response.read())
+    
+    def get_state_id(self):
+        status = self.get_status()
+        state_id = status["state_id"]
+        return state_id
+    
+    """ Choices are: {3, 4, 11}, State: {Ready, Pause, Manualcontrol}
+    """
+    def set_state_id(self, stateId):
+        return self.http.put("/status", json.dumps({'state_id': stateId}))
+    
+    def is_ready(self):
+        status = self.get_status()
+        if status["state_id"] != 3: # 3=Ready, 4=Pause, 11=Manualcontrol
+            self.logger.warn("MIR currently occupied. System state: {}".format(status["state_text"]))
+            return False
+        else:
+            return True
+    
+    def get_all_settings(self, advanced=False, listGroups=False):
+        if advanced:
+            response = self.http.get("/settings/advanced")
+        elif listGroups:
+            response = self.http.get("/setting_groups")
+        else:
+            response = self.http.get("/settings")
+        return json.loads(response.read())
+    
+    def get_group_settings(self, groupID):
+        response = self.http.get("/setting_groups/" + groupID + "/settings")
+        return json.loads(response.read())
+    
+    def set_setting(self, settingID, settingData):
+        return self.http.put("/setting", json.dumps({settingID: settingData}))
+    
+    def sync_time(self):
+        timeobj = datetime.now()
+        dT = timeobj.strftime("%Y-%m-%dT%X")
+        response = 'REST API: '
+        try:
+            response += str(self.http.put("/status", json.dumps({'datetime': dT})))
+        except Exception as e:
+            if str(e) == "timed out":
+                # setting datetime over REST API seems not to be intended
+                # that's why there is no response accompanying the PUT request,
+                # therefore a time out occurs, however time has been set correctly
+                response += "Set datetime to " + dT
+                self.logger.info("REST API: Setting time Mir triggers emergency stop, please unlock.")
```

Comment:
Lets make this a warn message, so one can easily see it since the GET failed messages afterwards flush the log output